lxd/instance/drivers/driver_qemu.go

```diff
@@ -1267,6 +1270,144 @@ echo "To start it now, unmount this filesystem and run: systemctl start lxd-agen
 		return err
 	}
 
+	// Templated files.
+	err = os.MkdirAll(filepath.Join(configDrivePath, "files"), 0500)
+	if err != nil {
+		return err
+	}
+
+	// Template anything that needs templating.
+	key := "volatile.apply_template"
+	if vm.localConfig[key] != "" {
+		// Run any template that needs running.
+		err = vm.templateApplyNow(vm.localConfig[key], filepath.Join(configDrivePath, "files"))
+		if err != nil {
+			return err
+		}
+
+		// Remove the volatile key from the DB.
+		err := vm.state.Cluster.ContainerConfigRemove(vm.id, key)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = vm.templateApplyNow("start", filepath.Join(configDrivePath, "files"))
+	if err != nil {
+		return err
+	}
+
+	// Copy the template metadata itself too.
+	metaPath := filepath.Join(vm.Path(), "metadata.yaml")
+	if shared.PathExists(metaPath) {
+		err = shared.FileCopy(metaPath, filepath.Join(configDrivePath, "files/metadata.yaml"))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (vm *qemu) templateApplyNow(trigger string, path string) error {
+	// If there's no metadata, just return.
+	fname := filepath.Join(vm.Path(), "metadata.yaml")
+	if !shared.PathExists(fname) {
+		return nil
+	}
+
+	// Parse the metadata.
+	content, err := ioutil.ReadFile(fname)
+	if err != nil {
+		return errors.Wrap(err, "Failed to read metadata")
+	}
+
+	metadata := new(api.ImageMetadata)
+	err = yaml.Unmarshal(content, &metadata)
+	if err != nil {
+		return errors.Wrapf(err, "Could not parse %s", fname)
+	}
+
+	// Figure out the instance architecture.
+	arch, err := osarch.ArchitectureName(vm.architecture)
+	if err != nil {
+		arch, err = osarch.ArchitectureName(vm.state.OS.Architectures[0])
+		if err != nil {
+			return errors.Wrap(err, "Failed to detect system architecture")
+		}
+	}
+
+	// Generate the container metadata
```

Comment:
@stgraber just a nit, missing `.` at end of comment.

Reply:
Fixed